pennylane/pauli/conversion.py

```diff
@@ -215,13 +239,158 @@ def _generalized_pauli_decompose(
+            A tuple ``(coeffs, terms)`` where ``coeffs`` is a complex-valued array of coefficients.
+            ``terms`` is either a list of operator tensors (if ``pauli=False``) or a list of
+            lists of ``(pauli_char, wire)`` pairs (if ``pauli=True``).
+
+    Raises:
+        ValueError: If the input has the wrong shape (not square or not a power of two when
+            ``padding=False``), or if the matrix is empty.
+
+    Example:
+        >>> import pennylane as qml
+        >>> import scipy.sparse as sp
+        >>> sparse_matrix = sp.csr_matrix([[0, 1], [1, 0]])
+        >>> coeffs, terms = qml.pauli.conversion._generalized_pauli_decompose_sparse(
+        ...     sparse_matrix, wire_order=[0]
+        ... )
+        >>> coeffs
+        array([1.+0.j])
+        >>> terms
+        [X(0)]
+    """
+    sparse_matrix = sp.coo_matrix(matrix)
+    sparse_matrix.sum_duplicates()
+    shape = sparse_matrix.shape
+
+    if padding:
+        max_dim = max(shape)
+        if max_dim == 0:
+            target_dim = 1
+        else:
+            target_dim = 2 ** math.ceil(math.log2(max_dim))
+        if shape != (target_dim, target_dim):
+            sparse_matrix = sp.coo_matrix(
+                (sparse_matrix.data, (sparse_matrix.row, sparse_matrix.col)),
+                shape=(target_dim, target_dim),
+            )
+            shape = sparse_matrix.shape
+
+    if shape[0] != shape[1]:
+        raise ValueError(
+            f"The matrix should be square, got {shape}. Use 'padding=True' for rectangular matrices."
+        )
+
+    if shape[0] == 0:
+        raise ValueError("Cannot decompose an empty matrix.")
+
+    num_qubits = int(math.log2(shape[0]))
+    if shape[0] != 2**num_qubits:
+        raise ValueError(
+            f"Dimension of the matrix should be a power of 2, got {shape}. Use 'padding=True' for these matrices."
+        )
+
+    if wire_order is not None and len(wire_order) != num_qubits:
+        raise ValueError(
+            f"number of wires {len(wire_order)} is not compatible with the number of qubits {num_qubits}"
+        )
+    if wire_order is None:
+        wire_order = range(num_qubits)
+
+    coeffs_map: dict[str, complex] = defaultdict(complex)
+    rows, cols, data = sparse_matrix.row, sparse_matrix.col, sparse_matrix.data
+
+    for row, col, value in zip(rows, cols, data):
+        if value == 0:
+            continue
+        contributions = [("", complex(value))]
+        for wire in range(num_qubits):
+            bit_index = num_qubits - 1 - wire
+            row_bit = (row >> bit_index) & 1
+            col_bit = (col >> bit_index) & 1
+            if row_bit == col_bit:
+                z_coeff = 0.5 if row_bit == 0 else -0.5
+                options = (("I", 0.5), ("Z", z_coeff))
+            else:
+                if row_bit == 0:
+                    options = (("X", 0.5), ("Y", 0.5j))
+                else:
+                    options = (("X", 0.5), ("Y", -0.5j))
```

Comment:
Optional comment to reduce the nestedness of this code block. Feel free to ignore.
```suggestion
            if row_bit == col_bit:
                options = (("I", 0.5), ("Z", (-1) ** row_bit * 0.5))
            else:
                options = (("X", 0.5), ("Y", (-1) ** row_bit * 0.5j))
```